# Import the necessary modules

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score




# Load the Dataset

In [2]:
# Load the dataset from the original source
data_url = "http://lib.stat.cmu.edu/datasets/boston"
raw_df = pd.read_csv(data_url, sep="\s+", skiprows=22, header=None)
data = np.hstack([raw_df.values[::2, :], raw_df.values[1::2, :2]])
target = raw_df.values[1::2, 2]



# Preprocessing and splitting the data into test and train datasets

In [9]:
# Step 2: Preprocess the data (no specific preprocessing needed for this example)





In [4]:
# Step 3: Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(data, target, test_size=0.2, random_state=42)

# Choose a model

In [5]:
# Step 4: Choose a regression model and train it on the training data
model = LinearRegression()
model.fit(X_train, y_train)



LinearRegression()

# Evaluate the Model

In [6]:
# Step 5: Evaluate the model on the testing data
y_pred = model.predict(X_test)



# Evaluation Metrics

In [8]:
# Calculate evaluation metrics
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print("Mean Squared Error:", mse)
print("R-squared:", r2)


Mean Squared Error: 24.291119474973566
R-squared: 0.6687594935356314


R-squared, also known as the coefficient of determination, is a measure that tells us how well the independent variable(s) in a regression model explain the variation in the dependent variable. It is a value between 0 and 1, where 0 means the model explains none of the variation, and 1 means the model explains all the variation.
 the model explains approximately 66.88% of the variability in the data, which is relatively good but may still leave some room for improvement.
 MSE is a way to measure how well a mathematical model, like a regression model, predicts or approximates actual data points. It calculates the average of the squared differences between the predicted values and the actual values. In simpler terms, it tells us how spread out the errors are between the predicted values and the real data points. The lower the MSE, the better the model is at making accurate predictions.